### PR TITLE
HDDS-6185: EC: EC keys can't be created via S3 interfaces

### PR DESCRIPTION
**Pull request by umamaheswararao**
## What changes were proposed in this pull request?

Handled replication config in put API.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6185

## How was this patch tested?

Added tests to cover.

### CODE REVIEW

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java [hunk 1]
```diff
@@ -41,4 +49,71 @@ public static String urlEncode(String str)
   private S3Utils() {
     // no instances
   }
+
+  /**
+   * This API used to resolve the client side configuration preference for file
+   * system layer implementations.
+   *
+   * @param s3StorageTypeHeader                - s3 user passed storage type
```
Comment (sodonnel) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java:57]: Formatting in this comment is a little out of sync with the others.

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java [hunk 2]
```diff
@@ -41,4 +49,71 @@ public static String urlEncode(String str)
...
+    if (s3StorageTypeHeader != null && !s3StorageTypeHeader.equals("")) {
+      s3StorageType = toS3StorageType(s3StorageTypeHeader);
+    }
+
+    boolean isECBucket = bucketReplConfig != null && bucketReplConfig
+        .getReplicationType() == HddsProtos.ReplicationType.EC;
+
+    // if bucket replication config configured with EC, we will give high
+    // preference to server side bucket defaults.
+    // Why we give high prefernce to EC is, there is no way for file system
```
Comment (sodonnel) [inline hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java:83]: Typo prefernce -> preference

**Comment by sodonnel** [comment]
These changes look good to me - I just spotted a couple of minor typos in the comments that are worth fixing.

Will there be other Jiras needed for other parts of the S3 interface, eg multi-part upload?

**Comment by umamaheswararao** [comment]
Thanks @sodonnel for the review. I have address the review comments above and pushed a commit. Thanks

**Comment by umamaheswararao** [comment]
Thanks @sodonnel for the reviews.!!!

**Pull request merged**